src/test/java/cmpt276/pg6/realtorest/controllers/MainControllerTest.java

```diff
@@ -32,4 +32,79 @@ static void setUp() {
     void testShowHomePage() throws Exception {
         this.mockMvc.perform(MockMvcRequestBuilders.get("/")).andExpect(MockMvcResultMatchers.status().isOk()).andExpect(MockMvcResultMatchers.view().name("home"));
     }
+
+    @Test
+    void testShowLoginPage() throws Exception {
+        this.mockMvc.perform(MockMvcRequestBuilders.get("/login")).andExpect(MockMvcResultMatchers.status().isOk()).andExpect(MockMvcResultMatchers.view().name("users/login"));
+    }
+    @Test
+    void testShowRegisterPage() throws Exception {
+        this.mockMvc.perform(MockMvcRequestBuilders.get("/register")).andExpect(MockMvcResultMatchers.status().isOk()).andExpect(MockMvcResultMatchers.view().name("users/register"));
+    }
+    @Test
+    void testShowPropertyListingsPage() throws Exception {
+        this.mockMvc.perform(MockMvcRequestBuilders.get("/property-listings"))
+        .andExpect(MockMvcResultMatchers.status().isOk())
+        .andExpect(MockMvcResultMatchers.view().name("propertyListings"));
+    }
+
+    @Test
+    void testUserLogin() throws Exception {
+        this.mockMvc.perform(MockMvcRequestBuilders.post("/login")
+        .param("email", "email@example.com")
+        .param("password", "password12"))
+        .andExpect(MockMvcResultMatchers.status().isAccepted())
+        .andExpect(MockMvcResultMatchers.redirectedUrl("/"));
+    }
+
+    @Test
+    void testUserLogout() throws Exception {
+        this.mockMvc.perform(MockMvcRequestBuilders.get("/logout"))
+        .andExpect(MockMvcResultMatchers.status().isOk())
```

Comment:
This assertion can't be passed because `java.lang.AssertionError: Status expected:<200> but was:<302>`.

Not sure if 302 response is the intended behavior, but this test needs to adjust.

Also, should there be a test to see if the session cookie can be destroyed?